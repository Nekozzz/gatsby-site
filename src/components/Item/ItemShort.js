import React from "react";
import { Link } from "gatsby"

const ItemShort = ({car}) => {

  return (
    <div key={car.slug}>
      <div>
        {car && (
          <img className={'img-preview'} src={`http://lorempixel.com/460/230/transport/${car.id}/`} alt=""/>
        )}
      </div>
      <div>
        Марка: {car.car_brand.brand_name}
      </div>
      <div>
        Модель: {car.model_name}
      </div>
      <Link to={`/${car.car_brand.slug}/${car.slug}`}>Узнать больше...</Link>
      <hr  style={{
        'marginTop': '8px',
        'marginBottom': '24px'
      }} />
    </div>
  )
};

export default ItemShort
