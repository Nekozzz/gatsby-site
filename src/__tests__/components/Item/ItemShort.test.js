import React from "react"
import renderer from "react-test-renderer"
import ItemShort from "../../../components/Item/ItemShort";

describe("ItemShort", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<ItemShort car={
        {
          car_brand: {
              brand_name: "Ford",
              slug: "ford"
          },
          id: 2,
          model_name: "Ford Focus III",
          slug: "ford-focus-iii",
        }
      }/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})