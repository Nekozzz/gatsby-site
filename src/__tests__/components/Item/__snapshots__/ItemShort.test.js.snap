// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemShort renders correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      className="img-preview"
      src="http://lorempixel.com/460/230/transport/2/"
    />
  </div>
  <div>
    Марка: 
    Ford
  </div>
  <div>
    Модель: 
    Ford Focus III
  </div>
  <a
    href="/ford/ford-focus-iii"
  >
    Узнать больше...
  </a>
  <hr
    style={
      Object {
        "marginBottom": "24px",
        "marginTop": "8px",
      }
    }
  />
</div>
`;
